### PR TITLE
Bug 1511821 - Replace old .then() syntax with async/await

### PR DESCRIPTION
**Pull request by ionutgoldan**

### CODE REVIEW

**Comment by edmorley** [comment]
For frontend changes like these, Cameron/Sarah will be better reviewers than I :-)

File: ui/perfherder/helpers.js [hunk 1]
```diff
@@ -673,16 +664,20 @@ export const getTitle = alertSummary => {
...
+};
 
-export const unassignBug = alertSummary =>
-  update(getApiUrl(`/performance/alertsummary/${alertSummary.id}/`), {
+export const unassignBug = async alertSummary => {
+  await update(getApiUrl(`/performance/alertsummary/${alertSummary.id}/`), {
     bug_number: null,
-  }).then(() => refreshAlertSummary(alertSummary));
+  });
+  await refreshAlertSummary(alertSummary);
```
Comment (sarah-clements) [inline ui/perfherder/helpers.js]: Does this `await` need to be here? Presumably, it just needs to wait for line 676's promise to return before executing this function?

File: ui/perfherder/helpers.js [hunk 2]
```diff
@@ -702,27 +697,29 @@ export const getAlertSummaryStatusText = alertSummary =>
...
-export const getAlertSummaryTitle = id =>
-  getAlertSummary(id).then(alertSummary => getTitle(alertSummary));
+  return AlertSummary(data, optionCollectionMap);
+};
 
-export const getAlertSummaries = options => {
+export const getAlertSummaryTitle = async id => {
+  const alertSummary = await getAlertSummary(id);
+  return getTitle(alertSummary);
+};
```
Comment (sarah-clements) [inline ui/perfherder/helpers.js]: I'm not sure there's much benefit in switching this small function to async/await. I think its more useful when there are multiple nested .then()'s :)

File: ui/perfherder/helpers.js [hunk 3]
```diff
@@ -615,17 +608,15 @@ export const isResolved = alertSummary =>
...
-        );
-      }),
+export const refreshAlertSummary = async alertSummary => {
+  const { data } = await getData(
+    getApiUrl(`/performance/alertsummary/${alertSummary.id}/`),
   );
+  const optionCollectionMap = await OptionCollectionModel.getMap();
+
+  Object.assign(alertSummary, data);
+  alertSummary.alerts = getInitializedAlerts(alertSummary, optionCollectionMap);
```
Comment (sarah-clements) [inline ui/perfherder/helpers.js]: Does `alertSummary` need to be returned?

File: ui/js/controllers/perf/alerts.js [hunk 4]
```diff
@@ -636,7 +626,7 @@ perf.controller('AlertsCtrl', [
                 $scope.frameworks = frameworks;
             }), OptionCollectionModel.getMap().then(function (optionCollectionMap) {
                 $scope.optionCollectionMap = optionCollectionMap;
-            })]).then(function () {
+            })]).then(async () => {
```
Comment (sarah-clements) [inline ui/js/controllers/perf/alerts.js]: Some of these nested .then's() could also be replaced with await.

File: ui/perfherder/helpers.js (hunk 2 above)
Comment (camd) [reply ui/perfherder/helpers.js]: +1

File: ui/perfherder/helpers.js [hunk 5]
```diff
@@ -759,36 +756,38 @@ export const getAlertSummaries = options => {
...
+  });
+  response = await response.json();
+
+  const newAlertSummaryId = response.alert_summary_id;
+  await create(getApiUrl('/performance/alert/'), {
+    summary_id: newAlertSummaryId,
+    signature_id: data.series.id,
+  });
+
+  return newAlertSummaryId;
```
[the remark this answers is not in the document]
Comment (ionutgoldan) [reply ui/perfherder/helpers.js]: Yes, otherwise the flow of the promise chain gets broken.

**Comment by edmorley** [comment]
What's the status of this PR? :-)

**Comment by ionutgoldan** [comment]
Well, I didn't merge this on time. The conflicts aren't too manageable now because of that.

**Comment by ionutgoldan** [comment]
Closing this as per [Bugzilla comment](https://bugzilla.mozilla.org/show_bug.cgi?id=1511821#c3).

**Pull request closed without merging**